scripts/proto-update-deps.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+set -e
+
+#
+# Download third_party proto files from the versions declared in go.mod
+#
+
+dir="$(pwd)"
+EXT_PROTO_DIR="$dir"/third_party
+
+# clean
+#rm -rf "${EXT_PROTO_DIR:?}/"*
+
+# Retrieve versions from go.mod (single source of truth)
+CONFIO_PROTO_URL=https://raw.githubusercontent.com/confio/ics23/$(go list -m github.com/confio/ics23/go | sed 's:.* ::')/proofs.proto
+GOGO_PROTO_URL=https://raw.githubusercontent.com/regen-network/protobuf/$(go list -m github.com/gogo/protobuf | sed 's:.* ::')/gogoproto/gogo.proto
+COSMOS_PROTO_URL=https://raw.githubusercontent.com/regen-network/cosmos-proto/master/cosmos.proto
+COSMWASM_TARBALL_URL=github.com/CosmWasm/wasmd/tarball/v0.17.0  # Backwards compatibility. Needed to serialize/deserialize older wasmd protos.
+WASMD_TARBALL_URL=$(go list -m github.com/CosmWasm/wasmd | sed 's:.* => ::' | sed 's/ /\/tarball\//')
+IBC_GO_TARBALL_URL=$(go list -m github.com/cosmos/ibc-go/v2 | sed 's:.* => ::' | sed 's/\/v2//' | sed 's/ /\/tarball\//')
+COSMOS_TARBALL_URL=$(go list -m github.com/cosmos/cosmos-sdk | sed 's:.* => ::' | sed 's/ /\/tarball\//')
+TM_TARBALL_URL=$(go list -m github.com/tendermint/tendermint | sed 's:.* => ::' | sed 's/ /\/tarball\//')
+
+# Download third_party protos
+mkdir -p "$EXT_PROTO_DIR"/proto || exit 1
+#cp -r "$dir"/third_party/proto/google "$EXT_PROTO_DIR"/proto || exit 1
+cd "$EXT_PROTO_DIR" || exit 1
+PROTO_EXPR="*/proto/**/*.proto"
+curl -sSL "$CONFIO_PROTO_URL" -o proto/proofs.proto.orig --create-dirs || exit 1
+curl -sSL "$GOGO_PROTO_URL" -o proto/gogoproto/gogo.proto --create-dirs || exit 1
+curl -sSL "$COSMOS_PROTO_URL" -o proto/cosmos_proto/cosmos.proto --create-dirs || exit 1
+curl -sSL "$COSMWASM_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" "$PROTO_EXPR" || exit 1
+curl -sSL "$COSMWASM_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" "$PROTO_EXPR" || exit 1
+curl -sSL "$WASMD_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" --exclude="*/proto/ibc" "$PROTO_EXPR" || exit 1
+curl -sSL "$IBC_GO_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" "$PROTO_EXPR" || exit 1
+curl -sSL "$COSMOS_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" --exclude="*/testutil" "$PROTO_EXPR" || exit 1
+curl -sSL "$TM_TARBALL_URL" | tar zx --strip-components 1 --exclude="*/third_party" "$PROTO_EXPR" || exit 1
```

Comment:
All of this comment is just nit stuff and bash tips.

You've got `set -e` at the top, so you don't need the `|| exit 1` on any of these lines. Also, unless there's a reason to have a specific exit code, doing `|| exit $?` is more desirable. The `$?` variable is the exit code of the last thing executed. Changing it to `set -ex` might be nice too. the `-x` will cause it to output commands as it executes them. That gives a nice paper-trail for when things go wrong (as well as making it easy to see what it's doing). Alternatively, there's the `-v` flag which will also output commands as it executes them, but it doesn't expand the variables in the commands and generally outputs a little less stuff. But `-x` is closer to default behavior from `make`.

Because of line 8: `dir="$(pwd)"`, this assumes you're running the script from the root of the repo. That's fine because we'll probably be running this using `make` which means we'd be in the repo root. But if you wanted it to work from any directory, you could do this:
```
dir="$( cd "$( dirname "${BASH_SOURCE:-$0}" )/.."; pwd -P )"
```
That then assumes that this script file is in the `scripts/` directory (or really 1 directory inside the root of the repo). But executing the script from anywhere (e.g. while currently in the `scripts/` directory) would then always update the desired location.

Reply:
Thank You! This is great. 

`dir="$( cd "$( dirname "${BASH_SOURCE:-$0}" )/.."; pwd -P )"`. This will help when running protoBindings for java/kotling